tests/cl-go-client/main.go

```diff
@@ -231,6 +238,78 @@ func swapGivenLargeAmountsBothDirections(igniteClient cosmosclient.Client, poolI
 	log.Println("finished swapping, num swaps done", numSwaps)
 }
 
+func createExternalCLIncentive(igniteClient cosmosclient.Client, poolId uint64, gaugeCoins sdk.Coins, expectedEpochIdentifier string) {
+	var (
+		randAccountNum = rand.Intn(8) + 1
+		accountName    = fmt.Sprintf("%s%d", accountNamePrefix, randAccountNum)
+	)
+
+	epochsQueryClient := epochstypes.NewQueryClient(igniteClient.Context())
+	currentEpochResponse, err := epochsQueryClient.CurrentEpoch(context.Background(), &epochstypes.QueryCurrentEpochRequest{
+		expectedEpochIdentifier,
+	})
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	log.Println("current epoch", currentEpochResponse.CurrentEpoch, "epoch identifier", expectedEpochIdentifier)
+
+	log.Println("querying epoch info. Note that incentives are distributed at the end of epoch. That's why it matters.")
+	epochInfosRespose, err := epochsQueryClient.EpochInfos(context.Background(), &epochstypes.QueryEpochsInfoRequest{})
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	if len(epochInfosRespose.Epochs) > 0 {
+		lastEpochInfo := epochInfosRespose.Epochs[len(epochInfosRespose.Epochs)-1]
+		log.Println("epoch duration", lastEpochInfo, "next epoch start time", lastEpochInfo.StartTime.Add(lastEpochInfo.Duration))
+	} else {
+		log.Println("could not find information about previous epoch. If duration is too long, this test might be infeasible")
+	}
+
+	//.Create gauge
+	runMessageWithRetries(func() error {
+		return createGauge(igniteClient, expectedPoolId, accountName, gaugeCoins)
+	})
+
+	epochAfterGaugeCreation := int64(-1)
+	for {
+		// Wait for 1 epoch to pass
+		currentEpochResponse, err = epochsQueryClient.CurrentEpoch(context.Background(), &epochstypes.QueryCurrentEpochRequest{
+			expectedEpochIdentifier,
+		})
+		if err != nil {
+			log.Fatal(err)
+		}
+		if epochAfterGaugeCreation == -1 {
+			log.Println("current epoch after gauge creation", currentEpochResponse.CurrentEpoch)
+			log.Println("waiting for next epoch...")
```

Comment:
does this get printed for an hour?

Reply:
maybe we can set a timer and retry every minute

Reply:
On localosmosis, unintuitively, an "hour" epoch is actually 60 seconds so this shouldn't be a concern.

If this is used on testnet, there is no "hour" epoch so the script would fail early.

Thi script is mainly used for testing incentives on localosmosis. If there is an infinite loop somewhere, the user can just kill the script

Reply:
Ah I see, now I understand why everything was so confusing while reviewing this, now everything became more clearer